src/onboarding/steps.test.ts

```diff
@@ -0,0 +1,237 @@
+import { BIOMETRY_TYPE } from 'react-native-keychain'
+import { initializeAccount } from 'src/account/actions'
+import { setHasSeenVerificationNux } from 'src/identity/actions'
+import {
+  navigate,
+  navigateClearingStack,
+  navigateHome,
+  popToScreen,
+} from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import {
+  END_OF_ONBOARDING_SCREEN,
+  FIRST_ONBOARDING_SCREEN,
+  getOnboardingStepValues,
+  goToNextOnboardingScreen,
+  _getScreenInfo,
+} from 'src/onboarding/steps'
+import { store } from 'src/redux/store'
+import { mockOnboardingProps } from 'test/values'
+import { mocked } from 'ts-jest/utils'
+
+jest.mock('src/redux/store', () => ({ store: { dispatch: jest.fn() } }))
+
+const mockStore = mocked(store)
+
+describe('onboarding steps', () => {
+  describe('_getScreenInfo and getOnboardingStepValues', () => {
+    it('All paths from FIRST_ONBOARDING_SCREEN lead to END_OF_ONBOARDING_SCREEN and have incrementing steps', () => {
+      const options = {
+        recoveringFromStoreWipe: [true, false],
+        choseToRestoreAccount: [true, false],
+        supportedBiometryType: [null, BIOMETRY_TYPE.FACE_ID],
+        skipVerification: [true, false],
+        numberAlreadyVerifiedCentrally: [true, false],
+      }
+
+      options.recoveringFromStoreWipe.forEach((recoveringFromStoreWipe) => {
+        options.choseToRestoreAccount.forEach((choseToRestoreAccount) => {
+          options.supportedBiometryType.forEach((supportedBiometryType) => {
+            options.skipVerification.forEach((skipVerification) => {
+              options.numberAlreadyVerifiedCentrally.forEach((numberAlreadyVerifiedCentrally) => {
+                // Initiate Test Variables
+                const onboardingProps = {
+                  recoveringFromStoreWipe,
+                  choseToRestoreAccount,
+                  supportedBiometryType,
+                  skipVerification,
+                  numberAlreadyVerifiedCentrally,
+                }
+                let currentScreen: Screens = FIRST_ONBOARDING_SCREEN
```

Comment:
this test looks like it largely duplicates the implementation of getOnboardingStepValues, and then checks that the answer is the same at the end. This seems too circular to me. Also, it's complex enough that there could easily be bugs in the test itself.

Reply:
one way you could write this test is select a subset of cases for your test and hardcoding the expected values (after checking them manually). But computing them on the fly like this with essentially the same code is going to make the test less reliable, harder to maintain, and less useful overall